codalab/rest/bundle.py

```diff
@@ -1,19 +1,346 @@
+    worksheet_util.check_worksheet_not_frozen(worksheet)
+    request.user.check_quota(need_time=True, need_disk=True)
+
+    created_uuids = []
+    for bundle in bundles:
+        # Prep bundle info for saving into database
+        # Unfortunately cannot use the `construct` methods because they don't
+        # provide a uniform interface for constructing bundles for all types
+        # Hopefully this can all be unified after REST migration is complete
+        bundle_uuid = spec_util.generate_uuid()
+        created_uuids.append(bundle_uuid)
+        bundle['uuid'] = bundle_uuid
+        bundle['owner_id'] = request.user.user_id
+        bundle['state'] = State.CREATED
+        bundle.setdefault('metadata', {})['created'] = int(time.time())
+        for dep in bundle.get('dependencies', []):
+            dep['child_uuid'] = bundle_uuid
+
+        # Create bundle object
+        bundle = get_bundle_subclass(bundle['bundle_type'])(bundle, strict=False)
+
+        # Save bundle into model
+        local.model.save_bundle(bundle)
+
+        # Inherit worksheet permissions
+        group_permissions = local.model.get_group_worksheet_permissions(
+            request.user.user_id, worksheet_uuid)
+        set_bundle_permissions([{
+            'object_uuid': bundle_uuid,
+            'group_uuid': p['group_uuid'],
+            'permission': p['permission'],
+        } for p in group_permissions])
+
+        # Add as item to worksheet
+        if shadow_parent_uuid is None:
+            local.model.add_worksheet_item(
+                worksheet_uuid, worksheet_util.bundle_item(bundle_uuid))
+        else:
+            local.model.add_shadow_worksheet_items(
+                shadow_parent_uuid, bundle_uuid)
+
+    # Get created bundles
+    bundles_dict = get_bundle_infos(created_uuids)
+
+    # Return bundles in original order
+    bundles = [bundles_dict[uuid] for uuid in created_uuids]
+    return BundleSchema(many=True).dump(bundles).data
+
+
+@patch('/bundles', apply=AuthenticatedPlugin())
+def _update_bundles():
+    """
+    Bulk update bundles.
+    """
+    bundle_updates = BundleSchema(
+        strict=True, many=True,
+        dump_only=UPDATE_RESTRICTED_FIELDS,
+    ).load(request.json, partial=True).data
+
+    # Check permissions
+    bundle_uuids = [b['uuid'] for b in bundle_updates]
+    check_bundles_have_all_permission(local.model, request.user, bundle_uuids)
+
+    # Update bundles
+    for update in bundle_updates:
+        bundle = local.model.get_bundle(update.pop('uuid'))
+        local.model.update_bundle(bundle, update)
+
+    # Get updated bundles
+    bundles_dict = get_bundle_infos(bundle_uuids)
+
+    # Create list of bundles in original order
+    updated_bundles = [bundles_dict[uuid] for uuid in bundle_uuids]
+
+    return BundleSchema(many=True).dump(updated_bundles).data
+
+
+@delete('/bundles', apply=AuthenticatedPlugin())
+def _delete_bundles():
```

Comment:
The latter is shared with LocalBundleClient, so has to be a separate function. I've been going back and forth a lot on how to resolve this naming conflict, what are your thoughts?